ssc/src/Pos/Ssc/Worker.hs

```diff
@@ -202,6 +201,16 @@ onNewSlotCommitment pm slotId@SlotId {..} sendCommitment
         sscProcessOurMessage (sscProcessCommitment pm comm)
         sendOurData sendCommitment CommitmentMsg comm siEpoch 0
 
+-- | Generate a random Opening.
+randomOpening :: IO Opening
+randomOpening = snd <$> secureRandCommitmentAndOpening
+  where
+    secureRandCommitmentAndOpening = runSecureRandom $ do
+        t       <- randomNumberInRange 3 10
+        n       <- randomNumberInRange (t*2-1) (t*2)
```

Comment:
That range does not look right :). It looks to me like the range only has two valid values.

Reply:
I suspect that its actually supposed to be
```
       n       <- randomNumberInRange (2 ^ (t -1)) (2 ^ t)
```
but then the value of `t` should be restricted more than it is.